sa/sa.go

```diff
@@ -636,12 +636,33 @@ func (ssa *SQLStorageAuthority) UpdateRegistration(ctx context.Context, reg core
 // NewPendingAuthorization stores a new Pending Authorization
 func (ssa *SQLStorageAuthority) NewPendingAuthorization(ctx context.Context, authz core.Authorization) (core.Authorization, error) {
 	var output core.Authorization
+
+	// Check if we can recycle an existing, pending authz.
+	if features.Enabled(features.ReusePendingAuthz) {
+		idJSON, err := json.Marshal(authz.Identifier)
+		if err != nil {
+			return output, err
+		}
+
+		pa, err := selectPendingAuthz(ssa.dbMap, "WHERE identifier = ?", idJSON)
+		if err == sql.ErrNoRows {
+			// No existing authz found, proceed to create one.
+		} else if err == nil {
+			// We found an authz, but we still need to fetch its challenges. To
+			// simplify things, just call GetAuthorization, which takes care of that.
+			return ssa.GetAuthorization(ctx, pa.ID)
```

Comment:
Can you increment a stat for this case? It might be interesting to know down the road and we have a similar stat (`ReusedValidAuthz`) for the valid authz reuse case.